openedx/core/djangoapps/credentials/tests/test_tasks.py

```diff
@@ -155,6 +155,16 @@ def test_auto_execution(self, mock_send):
         with freeze_time(datetime(2017, 5, 1, 3)):
             cert2 = GeneratedCertificateFactory(user=self.user, course_id='course-v1:edX+Test+22')
 
+        # Make a certificate outside of the command’s time range, but add a date
+        # override within the time range. The certificate should be included.
+        with freeze_time(datetime(2017, 4, 30, 0)):
+            cert3 = GeneratedCertificateFactory(user=self.user, course_id='course-v1:edX+Test+33')
+        with freeze_time(datetime(2017, 5, 1, 2)):
+            CertificateDateOverrideFactory(
+                generated_certificate=cert3,
+                overridden_by=self.user,
+            )
```

Comment:
One more small request, can you add one more scenario where the cert is deleted and still sent? That seems to be the primary reason we are looking in the history table. Is that a correct assumption?